### ISSUE TITLE
AlgoliaApiException: {"message":"Invalid value for \\\"removeWordsIfNoResults\\\" parameter

### ISSUE THREAD

**Issue by rhamianiuk** [algolia/algoliasearch-client-java#628]
- Algolia Client Version: 3.0.0
- Language Version:

### Description
```setRemoveWordsIfNoResults(RemoveWordsType removeWordsIfNoResults)``` seems not matching real REST API. The response from Algolia is 
```com.algolia.search.exceptions.AlgoliaApiException: {"message":"Invalid value for \\\"removeWordsIfNoResults\\\" parameter, expected firstWords, lastWords or none","status":400}```

Looks like RemoveWordsType has invalid values:
```  
REMOVE_LAST_WORDS("LastWords")
  REMOVE_FIRST_WORDS("FirstWords"),
  REMOVE_NONE("none"),
  REMOVE_ALLOPTIONAL("allOptional");
```

### Steps To Reproduce

Just call Algolia API from java client using 

```
query.setRemoveWordsIfNoResults(RemoveWordsType.REMOVE_ALLOPTIONAL)
```

**Comment by Ant-hem** [comment]
Hi @rhamianiuk! 

Thanks for reporting the issue. I pushed a PR to address it.

**Comment by rhamianiuk** [comment]
Hey @Ant-hem!
I've just built jar-artifacts from master and tried the same requests.
Here is what I used:
![image](https://user-images.githubusercontent.com/1923886/63687453-5e13d980-c80d-11e9-8bc8-ae0b47193638.png)
But endpoint still returns the same error:
```
com.algolia.search.exceptions.AlgoliaApiException: {"message":"Invalid value for \\\"removeWordsIfNoResults\\\" parameter, expected firstWords, lastWords or none","status":400}
```

The same error for `allOptional` and `lastWords`

Note that enabling `allOptional` in Algolia index does the trick. But API doesn't even mention this option is possible.
